src/core/bonusMinting.js

```diff
@@ -47,6 +58,161 @@ export type BonusPeriod = {|
   +startTimeMs: TimestampMs | number,
 |};
 
+export type ComputedBonusMinting = $ReadOnlyArray<BonusMintOverTime>;
+
+export type BonusMintOverTime = {|
+  +recipient: GraphNode,
+  // Recipient's bonus minting in each interval.
+  +bonusIntervals: $ReadOnlyArray<BonusInterval>,
+|};
+
+export type BonusInterval = {|
+  +interval: Interval,
+  // The amount of bonus Cred to mint in this interval.
+  +amount: number,
+|};
+
+export function computeBonusMinting(
+  wg: WeightedGraphT,
+  policies: $ReadOnlyArray<BonusPolicy>
+): ComputedBonusMinting {
+  const mintIntervals = _computeMintIntervals(wg);
+  return policies.map((policy) => {
+    const recipient = wg.graph.node(policy.address);
+    if (recipient == null) {
+      throw new Error(
+        `bonus recipient not in graph: ${NodeAddress.toString(policy.address)}`
+      );
+    }
+    return {
+      recipient,
+      bonusIntervals: _bonusIntervals(mintIntervals, policy.periods),
+    };
+  });
+}
+
+export function _bonusIntervals(
+  mintIntervals: $ReadOnlyArray<MintInterval>,
+  periods: $ReadOnlyArray<BonusPeriod>
+): $ReadOnlyArray<BonusInterval> {
+  let periodIndex = -1;
```

Comment:
I did it that way because initially, there's no matching period, so setting the `periodIndex = -1` matches the behavior of e.g. `Array.indexOf` which returns `-1` to mean "index not found". That said, you're right that there's an opportunity to clean this up. I renamed `periodIndex` to `nextIndex` and applied your change.